simulator/infra/instrcache/LRUCache.h

```diff
@@ -73,31 +73,26 @@ class LRUCache
         void erase( const Key& key)
         {
             auto data_it = data.find( key);
-            auto lru_it  = lru_hash.find( key);
-            assert ( ( data_it == data.end()) == ( lru_it == lru_hash.end()));
+            assert ( data_it == data.end());
             if ( data_it != data.end())
             {
                 free_list.emplace_front( data_it->second);
                 storage[data_it->second].~Value();
                 data.erase( data_it);
-                lru_list.erase( lru_it->second);
```

Comment:
Here you should call a method to put the deleted element to the LRU position (something opposite to `touch`), so erased entry would be used in the 1st priority.